docs/docs/features/soft-off.md

```diff
@@ -0,0 +1,255 @@
+
+You can then wake up the keyboard by pressing the reset button once, and repeating this for each side for split keyboards.
+
+## Adding Dedicated Soft On/Off GPIO Pin To New Designs
+
+### Hardware Design
+
+ZMK's dedicated soft on/off pin feature requires a dedicated GPIO pin to be used to trigger powering off, and to wake the core from the
+soft off state when it goes active again later.
+
+#### Simple Direct Pin
+
+The simplest way to achieve this is with a push button between a GPIO pin and ground.
+
+#### Matrix-Integrated Hardware Combo
+
+Another, more complicated option is to tie two of the switch outputs in the matrix together through an AND gate and connect that to the dedicated GPIO pin. This way you can use a key combination in your existing keyboard matrix to trigger soft on/off. To make this work best, the two switches used should both be driven by the same matrix input pin so that both will be active simultaneously on the AND gate inputs. The alternative is to connect the switch to two MOSFETs that trigger both the regular matrix connect and the connect to the AND gate to ensure both pins are active/high at the same time even if scanning sets them high at different times.
+
+### Firmware Changes
+
+Several items work together to make both triggering soft off properly, and setting up the device to _wake_ from soft off work as expected. In addition, some small changes are needed to keep the regular idle deep sleep functionality working.
+
+#### Wakeup Sources
+
+Zephyr has general support for the concept of a device as a "wakeup source", which ZMK has not previously used. Adding soft off requires properly updating the existing `kscan` devices with the `wakeup-source` property to ensure they will still work to wake the device from regular inactive deep sleep, e.g.:
+
+```
+/ {
+    kscan0: kscan_0 {
+        compatible = "zmk,kscan-gpio-matrix";
+        label = "KSCAN";
+        diode-direction = "col2row";
+        wakeup-source;
+
+        ...
+    };
+};
+```
+
+#### GPIO Key
+
+Zephyr's basic GPIO Key concept is used to configure the GPIO pin that will be used for both triggering soft off and waking the device later. Here is an example for a keyboard with a dedicated on/off push button that is a direct wire between the GPIO pin and ground:
+
+```
+/ {
+    keys {
+        compatible = "gpio-keys";
+        wakeup_key: wakeup_key {
+            gpios = <&gpio0 2 (GPIO_ACTIVE_LOW | GPIO_PULL_UP)>;
+        };
+    };
+};
+```
+
+GPIO keys are defined using child nodes under the `gpio-keys` compatible node. Each child needs just one property defined:
+
+- The `gpios` property should be a phandle-array with a fully defined GPIO pin and with the correct pull up/down and active high/low flags set. In the above example the soft on/off would be triggered by pulling the specified pin low, typically by pressing a switch that has the other leg connected to ground.
+
+#### Soft Off Waker
```

Comment:
I am having some trouble understanding: Is this still necessary if you follow the "Simple Direct Pin" section under sideband?

Reply:
So.. good question. *Yes*, but let me think if we could avoid it and just depend on the direct kscan itself to do the waking in that case.

Reply:
Ok, so did a little testing, and no, it isn't actually required in that case, you can directly use the small dedicated direct kscan as the soft off waker.... Let me think about how best to capture this in the docs.

Reply:
@caksoylar Ok, pushed docs tweaks and changed the ZMK Uno shield to reflect the updated recommended approach that adds the extra waker *only* when using the matrix integrated approach.